migrate/playbooks/rds-timescale-live-migration.md

```diff
@@ -0,0 +1,407 @@
+psql -h aws-rds-low-downtime-migration-db.c8tzn206yp6f.us-west-2.rds.amazonaws.com -p 5432 -U postgres -c "select 1"
+Password for user postgres:
+ ?column?
+----------
+        1
+(1 row)
+```
+
+## Performing data migration
+<GettingHelp />
+
+The migration process consists of the following steps:
+1. Set up a target database instance in Timescale.
+1. Prepare the source database for the live migration.
+1. Set up a replication slot and snapshot.
+1. Migrate roles and schema from source to target.
+1. Perform "Live migration".
+
+<StepOne />
+
+## 2. Prepare the source database for the live migration
+In order to replicate `UPDATE` and `DELETE` operations on tables in the source database,
+the tables must have either a primary key or `REPLICA IDENTITY`. Replica identity
+assists logical decoding in identifying the rows being modified. It defaults to
+using the table's primary key.
+
+If a table doesn't have a primary key, you'll have to manually set the replica identity.
+One option is to use a unique, non-partial, non-deferrable index that includes only
+columns marked as `NOT NULL`.
+
+```sql
+ALTER TABLE {table_name} REPLICA IDENTITY USING INDEX {_index_name};
+```
+
+If there's no primary key or viable unique index to use, you will have to set
+`REPLICA IDENTITY` to `FULL`. If you are expecting a large number of `UPDATE` or `DELETE`
+operations on the table, we do not recommend using `FULL`. For each `UPDATE` or `DELETE`
+statement, Postgres will have to read the whole table to find all matching rows
+slowing the replication process.
+
+```sql
+ALTER TABLE {table_name} REPLICA IDENTITY FULL;
+```
+
+## 3. Set up a replication slot and snapshot
+Once you're sure that the tables which will be affected by `UPDATE` and `DELETE`
+queries have `REPLICA IDENTITY` set, you will need to create a replication slot.
+
+Replication slots keep track of transactions (recorded in Write-Ahead Log files)
+in the source database that have not been streamed to the target database yet. We
+will use `pgcopydb` to create a replication slot in the source database.
+
+```sh
+pgcopydb follow \
+  --source "$SOURCE" \
+  --target "$TARGET" \
+  --fail-fast \
+  --plugin wal2json
+```
+
+This command is going to be active during most of the migration process. You can
+run it in the background or use terminal multiplexers like `screen` or `tmux`.
+
+The `follow` command sets up a replication slot in the source database to stream
+changes. These changes are held in "intermediate machine" on disk until "apply" is
+given. We will discuss about apply command in subsequent steps.
+
+Additionally, `follow` command exports a snapshot ID to `/tmp/pgcopydb/snapshot`.
+This ID can be utilized to migrate data that was in the database before the replication
+slot was created.
+
+## 4. Migrate roles and schema from source to target
+Before applying Change Data Capture (CDC) operations such as inserts, updates, and deletes
```

Comment:
Reword, removing "change data capture". It's mentioned once in this whole document, and IMO only adds to confusion.